### PR TITLE
chore: update pre-processed ontology descendants and mapping files

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
Signed-off-by: nayib-jose-gloria <ngloria@chanzuckerberg.com>

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/337

### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- Regenerated cell_type_ontology_mapping.py by running [compute_tissue_and_cell_type_mappings.ipynb](https://github.com/chanzuckerberg/single-cell-curation/blob/main/notebooks/compute_tissue_and_cell_type_mappings.ipynb)
- Regenerated tissue_ontology_mapping.py by running [compute_tissue_and_cell_type_mappings.ipynb](https://github.com/chanzuckerberg/single-cell-curation/blob/main/notebooks/compute_tissue_and_cell_type_mappings.ipynb)
- Moved development_stage_ontology_mapping.py to new directory, did NOT need to regen because there was not a development_stage ontology refresh. Same pinned version as schema 2.0.0
- commited the ontology mapping json outputs to a fixture directory and had the Python file load from JSON instead 
- consolidated ontology mappings into a constants file in a util/ontology_mappings directory
- Regenerated CELL_TYPE_DESCENDANTS by running [compute_tissue_and_cell_type_mappings.ipynb](https://github.com/chanzuckerberg/single-cell-curation/blob/main/notebooks/compute_tissue_and_cell_type_mappings.ipynb) and copying data into the JS source file
- Regenerated TISSUE_DESCENDANTS by running [compute_tissue_and_cell_type_mappings.ipynb](https://github.com/chanzuckerberg/single-cell-curation/blob/main/notebooks/compute_tissue_and_cell_type_mappings.ipynb) and copying data into the JS source file

## Notes for Reviewer

### CODE REVIEW

**Review by atolopko-czi — CHANGES_REQUESTED** [review]
Great refactoring!

"Requesting changes" in order to confirm that mappings were regenerated using the latest 3.0 ontology versions. It looks like the notebook in s-c-curation repo hardcodes the ontology files. Were those versions updated on a branch, yet to be committed?

The s-c-curation notebooks could probably be moved to this repo, since they are portal-specific. Since the (copied) owl_info.yml file is in this repo, the notebooks could read it to dynamically find the latest ontology owl file versions.

Finally, what are your thoughts on having the BE API return the mappins to the FE, rather than having to hardcode them in a *js file? Not suggesting we do that on this branch, but is it possible to do so and worth doing?

**Comment by atolopko-czi** [comment]
@nayib-jose-gloria Could you also make any updates need to the [Schema Update Process doc](https://docs.google.com/document/d/1R3a43eLVtWLOxgSu2HCMdD8QK5hddHjtY7uZJrC4VuE/edit)? TY!

**Comment by atolopko-czi** [comment]
> > Great refactoring!
> > "Requesting changes" in order to confirm that mappings were regenerated using the latest 3.0 ontology versions. It looks like the notebook in s-c-curation repo hardcodes the ontology files. Were those versions updated on a branch, yet to be committed?
>
> They were generated with the latest 3.0 ontology versions, and they were committed yesterday. Are you seeing an issue? https://github.com/chanzuckerberg/single-cell-curation/tree/main/cellxgene_schema_cli/cellxgene_schema/ontology_files

https://github.com/chanzuckerberg/single-cell-curation/blob/main/notebooks/compute_tissue_and_cell_type_mappings.ipynb

```

# Load CL, pinned for 2.0.0 schema.
cl_ontology = "https://github.com/obophenotype/cell-ontology/raw/v2021-08-10/cl.owl"

...
# Load UBERON, pinned for 2.0.0 schema.
uberon_ontology = "https://github.com/obophenotype/uberon/raw/v2021-07-27/uberon.owl"
```

**Comment by nayib-jose-gloria** [comment]
> > > Great refactoring!
> > > "Requesting changes" in order to confirm that mappings were regenerated using the latest 3.0 ontology versions. It looks like the notebook in s-c-curation repo hardcodes the ontology files. Were those versions updated on a branch, yet to be committed?
> > 
> > 
> > They were generated with the latest 3.0 ontology versions, and they were committed yesterday. Are you seeing an issue? https://github.com/chanzuckerberg/single-cell-curation/tree/main/cellxgene_schema_cli/cellxgene_schema/ontology_files
> 
> https://github.com/chanzuckerberg/single-cell-curation/blob/main/notebooks/compute_tissue_and_cell_type_mappings.ipynb
> 
> ```
> 
> # Load CL, pinned for 2.0.0 schema.
> cl_ontology = "https://github.com/obophenotype/cell-ontology/raw/v2021-08-10/cl.owl"
> 
> ...
> # Load UBERON, pinned for 2.0.0 schema.
> uberon_ontology = "https://github.com/obophenotype/uberon/raw/v2021-07-27/uberon.owl"
> ```

ahhhh didn't realize it was hardcoded in the script--ty for flagging! Will regen

File: tests/unit/backend/corpora/api_server/test_v1_dataset.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def test__enrich_development_stage_with_ancestors_expands_correctly(self):
         dataset = {"development_stage": [{"ontology_term_id": "HsapDv:0000008", "label": "Test"}]}
         Dataset.enrich_development_stage_with_ancestors(dataset)
         self.assertIn("development_stage_ancestors", dataset)
-        self.assertEqual(
+        self.assertCountEqual(
```
Comment (atolopko-czi) [inline tests/unit/backend/corpora/api_server/test_v1_dataset.py:173]: I can't get over [how bad](https://github.com/python/cpython/issues/71258) `assertCountEqual` is as a method name! 😆

**Review by atolopko-czi — APPROVED** [review]
LGTM! 

Has `scripts/compute_tissue_and_cell_type_mappings.ipynb` also been removed from single-cell-curation?

File: tests/unit/backend/corpora/api_server/test_v1_dataset.py (hunk 1 above)
Comment (nayib-jose-gloria) [reply tests/unit/backend/corpora/api_server/test_v1_dataset.py:173]: I know, I considered just doing something else here to avoid the confusing name 😆

**Comment by nayib-jose-gloria** [comment]
> LGTM!
> 
> Has `scripts/compute_tissue_and_cell_type_mappings.ipynb` also been removed from single-cell-curation?

it hasn't, but I'll make a follow-up PR

**Pull request merged**